inference-engine/src/transformations/src/transformations/common_optimizations/shuffle_channels_fusion.cpp

```diff
@@ -0,0 +1,138 @@
+    auto transpose = std::make_shared<ngraph::opset3::Transpose>(reshape_before, input2);
+    auto reshape_after = std::make_shared<ngraph::opset3::Reshape>(transpose, input3, false);
+
+    ngraph::matcher_pass_callback callback = [this, reshape_constants_check](pattern::Matcher& m) {
+        auto reshape_after = std::dynamic_pointer_cast<ngraph::opset3::Reshape>(m.get_match_root());
+        if (!reshape_after) {
+            return false;
+        }
+
+        auto transpose = std::dynamic_pointer_cast<ngraph::opset3::Transpose>(reshape_after->input_value(0).get_node_shared_ptr());
+        if (!transpose || transpose->get_output_target_inputs(0).size() != 1) {
+            return false;
+        }
+
+        auto reshape_before = std::dynamic_pointer_cast<ngraph::opset3::Reshape>(transpose->input_value(0).get_node_shared_ptr());
+        if (!reshape_before || reshape_before->get_output_target_inputs(0).size() != 1) {
+            return false;
+        }
+
+        auto reshape_after_constant = std::dynamic_pointer_cast<ngraph::opset3::Constant>(reshape_after->input_value(1).get_node_shared_ptr());
+        auto transpose_constant = std::dynamic_pointer_cast<ngraph::opset3::Constant>(transpose->input_value(1).get_node_shared_ptr());
+        auto reshape_before_constant = std::dynamic_pointer_cast<ngraph::opset3::Constant>(reshape_before->input_value(1).get_node_shared_ptr());
+        if (!transpose_constant || !reshape_before_constant || !reshape_after_constant) {
+            return false;
+        }
+
+        if (reshape_constants_check) {
+            auto reshape_before_values = reshape_before_constant->cast_vector<int64_t>();
+            auto reshape_after_values = reshape_after_constant->cast_vector<int64_t>();
+            for (size_t i = 0; i < reshape_before_values.size(); ++i) {
+                if ((reshape_before_values[i] == -1) || (reshape_before_values[i] == 0 && i != 0)) {
+                    return false;
+                }
+            }
+            for (size_t i = 0; i < reshape_after_values.size(); ++i) {
+                if ((reshape_after_values[i] == -1) || (reshape_after_values[i] == 0 && i != 0)) {
+                    return false;
+                }
+            }
+        }
+
+        auto p_shape_input = reshape_before->get_input_partial_shape(0);
+        auto p_shape_reshape_before = reshape_before->get_output_partial_shape(0);
+        auto p_shape_transpose = transpose->get_output_partial_shape(0);
+        auto p_shape_transpose_constant = transpose_constant->get_output_partial_shape(0);
+        auto p_shape_reshape_after = reshape_after->get_output_partial_shape(0);
+
+        if (p_shape_input.is_dynamic() || p_shape_reshape_before.is_dynamic() ||
+            p_shape_transpose.is_dynamic() || p_shape_reshape_after.is_dynamic() ||
+            p_shape_transpose_constant.is_dynamic()) {
+            return false;
+        }
```

Comment:
move this check to pattern using `patter::has_static_shape()` predicate.

Reply:
Thanks, done